google/auth/_default.py

```diff
@@ -0,0 +1,295 @@
+            wrong format.
+    """
+    with io.open(filename, 'r') as file_obj:
+        try:
+            info = json.load(file_obj)
+        except ValueError as exc:
+            raise exceptions.DefaultCredentialsError(
+                'File {} is not a valid json file.'.format(filename), exc)
+
+    # The type key should indicate that the file is either a service account
+    # credentials file or an authorized user credentials file.
+    credential_type = info.get('type')
+
+    if credential_type == _AUTHORIZED_USER_TYPE:
+        credentials = google.oauth2.credentials.Credentials(
+            None,
+            refresh_token=info['refresh_token'],
+            token_uri=_GOOGLE_OAUTH2_TOKEN_ENDPOINT,
+            client_id=info['client_id'],
+            client_secret=info['client_secret'])
+        # Authorized user credentials do not contain the project ID.
+        return credentials, None
+
+    if credential_type == _SERVICE_ACCOUNT_TYPE:
+        credentials = service_account.Credentials.from_service_account_info(
+            info)
+        return credentials, info.get('project_id')
+
+    else:
+        raise exceptions.DefaultCredentialsError(
+            'The file {file} does not have a valid type. '
+            'Type is {type}, expected one of {valid_types}.'.format(
+                file=filename, type=credential_type, valid_types=_VALID_TYPES))
+
+
+def _get_explicit_environ_credentials():
+    """Gets credentials from the GOOGLE_APPLICATION_CREDENTIALS environment
+    variable."""
+    explicit_file = os.environ.get(_CREDENTIALS_ENV)
+    if explicit_file is not None:
+        return _load_credentials_from_file(os.environ[_CREDENTIALS_ENV])
+    else:
+        return None, None
+
+
+def _get_gcloud_sdk_project_id(config_path):
+    """Gets the project ID from the Cloud SDK's configuration.
+
+    Args:
+        config_path (str): The path to the Cloud SDK's config directory,
+            for example ``~/.config/gcloud``.
+
+    Returns:
+        Optional[str]: The project ID.
+    """
+    config_file = os.path.join(config_path, _CLOUDSDK_ACTIVE_CONFIG_FILENAME)
+
+    if not os.path.exists(config_file):
+        return None
+
+    config = configparser.RawConfigParser()
+
+    try:
+        config.read(config_file)
+    except configparser.Error:
+        return None
+
+    if config.has_section(_CLOUDSDK_PROJECT_CONFIG_SECTION):
+        return config.get(
+            _CLOUDSDK_PROJECT_CONFIG_SECTION, _CLOUDSDK_PROJECT_CONFIG_KEY)
+
+
+def _get_gcloud_sdk_config_path():
+    """Returns the absolute path the the Cloud SDK's configuration directory.
+
+    Returns:
+        str: The Cloud SDK config path.
+    """
+    if _CLOUDSDK_CONFIG_ENV in os.environ:
+        return os.environ[_CLOUDSDK_CONFIG_ENV]
+
+    # Non-windows systems store this at ~/.config/gcloud
+    if os.name != 'nt':
+        return os.path.join(
+            os.path.expanduser('~'), '.config', _CLOUDSDK_CONFIG_DIRECTORY)
```

Comment:
No, it should be fixed shortly. (it already works live, I'm awaiting the fix for the sdk).